### PR TITLE
Issue 1146 Get rid of TempNupkgFile | In Progress

### PR DESCRIPTION
**Pull request by ogryniuk**
@sbespalov 

I finished the main part of the issue and would be glad to know if it is right before starting to work on tests.

### CODE REVIEW

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/nuget/NuspecExtractor.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    private final void convertToInputStreamAndStore(RepositoryPath path, Map mapToProcess)
+            throws ProviderImplementationException,
+                   ArtifactCoordinatesValidationException,
+                   IOException
+    {
+        RepositoryPath repositoryPath = path;
+        Map<String, String> checksum = mapToProcess;
+        InputStream completeChecksum;
+
+        for(Map.Entry<String, String> mapEntry: checksum.entrySet()){
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/nuget/NuspecExtractor.java]: Cosmetic: Braces on new lines, please. :)

**Comment by carlspring** [comment]
@ogryniuk ,

Could you please continue with the tests as well?

**Comment by ogryniuk** [comment]
@carlspring 

I sure will continue with the tests. 

Just really need to know first that the core of my commit is ok in itself

**Comment by carlspring** [comment]
@ogryniuk ,

I've had a look at your pull request and without meaning to offend you, I think it would be better, if you picked another task. This task requires at least some knowledge of Nuget. The `strongbox-storage-nuget-layout-provider` module has quite a few tests (which could be used as illustrations of how to do things), including such required to cover the functionality of generating Nuget test artifacts and resources.

It looks like you'll need a lot of help in order to implement this (instead of being able to mostly do it independently) and we just don't have many resources to guide and mentor you on NuGet, which is why I think that some other task matching your skill-set will probably be better for you at this time.

I hope you understand.

**Comment by ogryniuk** [comment]
@carlspring ,

No problem

**Comment by carlspring** [comment]
Thank you for your understanding, @ogryniuk ! :)

**Pull request closed without merging**